src/channel/ws_0_8.ts

```diff
@@ -0,0 +1,564 @@
+    return this.unsubscribe(subId, WSSubscriptions.NEW_HEADS);
+  }
+
+  /**
+   * subscribe to 'starknet events'
+   * * you can subscribe to this event multiple times and you need to manage subscriptions manually
+   */
+  public subscribeEventsUnmanaged(
+    fromAddress?: BigNumberish,
+    keys?: string[][],
+    blockIdentifier?: BlockIdentifier
+  ) {
+    const block_id = blockIdentifier ? new Block(blockIdentifier).identifier : undefined;
+    return this.sendReceive('starknet_subscribeEvents', {
+      ...{ from_address: fromAddress && toHex(fromAddress) },
+      ...{ keys },
+      ...{ block: block_id },
+    }) as Promise<SUBSCRIPTION_RESULT>;
+  }
+
+  /**
+   * subscribe to 'starknet events'
+   */
+  public async subscribeEvents(
+    fromAddress?: BigNumberish,
+    keys?: string[][],
+    blockIdentifier?: BlockIdentifier
+  ) {
+    if (this.subscriptions.get(WSSubscriptions.EVENTS)) return false;
+    // eslint-disable-next-line prefer-rest-params
+    const subId = (await this.subscribeEventsUnmanaged(fromAddress, keys, blockIdentifier))
+      .subscription_id;
+    this.subscriptions.set(WSSubscriptions.EVENTS, subId);
+    return subId;
+  }
+
+  /**
+   * Unsubscribe 'starknet events' subscription
+   */
+  public unsubscribeEvents() {
+    const subId = this.subscriptions.get(WSSubscriptions.EVENTS);
+    if (!subId) throw Error('There is no subscription ID for this event');
+    return this.unsubscribe(subId, WSSubscriptions.EVENTS);
+  }
+
+  /**
+   * subscribe to transaction status
+   * * you can subscribe to this event multiple times and you need to manage subscriptions manually
+   */
+  public subscribeTransactionStatusUnmanaged(
+    transactionHash: BigNumberish,
+    blockIdentifier?: BlockIdentifier
+  ) {
+    const transaction_hash = toHex(transactionHash);
+    const block_id = blockIdentifier ? new Block(blockIdentifier).identifier : undefined;
+    return this.sendReceive('starknet_subscribeTransactionStatus', {
+      transaction_hash,
+      ...{ block: block_id },
+    }) as Promise<SUBSCRIPTION_RESULT>;
+  }
+
+  /**
+   * subscribe to transaction status
+   */
+  public async subscribeTransactionStatus(
+    transactionHash: BigNumberish,
+    blockIdentifier?: BlockIdentifier
+  ) {
+    if (this.subscriptions.get(WSSubscriptions.TRANSACTION_STATUS)) return false;
+    const subId = (await this.subscribeTransactionStatusUnmanaged(transactionHash, blockIdentifier))
+      .subscription_id;
+    this.subscriptions.set(WSSubscriptions.TRANSACTION_STATUS, subId);
+    return subId;
+  }
+
+  /**
+   * unsubscribe 'transaction status' subscription
+   */
+  public async unsubscribeTransactionStatus() {
+    const subId = this.subscriptions.get(WSSubscriptions.TRANSACTION_STATUS);
+    if (!subId) throw Error('There is no subscription ID for this event');
+    return this.unsubscribe(subId, WSSubscriptions.TRANSACTION_STATUS);
+  }
+
+  /**
+   * subscribe to pending transactions (mempool)
```

Comment:
todo add params on this and rest that are missing

Reply:
I think it is auto add on doccs ? @penovicp